### PR TITLE
Resolve vertical branch conflicts

### PR DESCRIPTION
**Pull request by pchenthx1**
Conflicts against Horizontal manifest baseline of WW01 release

Tracked-On: OAM-105613
Signed-off-by: Chenthati, Pradeep <pradeepx.chenthati@intel.com>

### CODE REVIEW

**Comment by pchenthx1** [comment]
@basilchew @bnkoppad kindly review the PR

**Review by basilchew — COMMENTED** [review]
It looks like a lot of patches are removed.
Is it because the vertical patches have been ported back to the horizontal branch?

This PR is raised using OAM-105613, however, there is no further information in it.
Previously, for every vertical patch added, there is a horizontal Jira submitted to port back to horizontal branch and a revert ticket to remove the patch from vertical.

For traceability, can we duplicate all such revert Jira tickets to OAM-105613? They should then appear as duplicates in the OAM-105613 ticket. With this info, we will then be able to check if the correct patches are removed.

**Comment by pchenthx1** [comment]
> It looks like a lot of patches are removed. Is it because the vertical patches have been ported back to the horizontal branch?
> 
> This PR is raised using OAM-105613, however, there is no further information in it. Previously, for every vertical patch added, there is a horizontal Jira submitted to port back to horizontal branch and a revert ticket to remove the patch from vertical.
> 
> For traceability, can we duplicate all such revert Jira tickets to OAM-105613? They should then appear as duplicates in the OAM-105613 ticket. With this info, we will then be able to check if the correct patches are removed.

Yes Basil, these patches are present in Horizontal branch. I have linked relevant revert Jira tickets as duplicate to OAM-105613.
Please review

File: bsp_diff/caas/vendor/intel/mediasdk_c2/0004-Fix-for-Vts-MultiRelease-decoder-TCs.patch [hunk 1]
```diff
@@ -1,32 +0,0 @@
...
-
-diff --git a/c2_components/src/mfx_c2_decoder_component.cpp b/c2_components/src/mfx_c2_decoder_component.cpp
-index b5e3c44..4fe12ce 100755
---- a/c2_components/src/mfx_c2_decoder_component.cpp
-+++ b/c2_components/src/mfx_c2_decoder_component.cpp
-@@ -1314,6 +1314,8 @@ void MfxC2DecoderComponent::DoConfig(const std::vector<C2Param*> &params,
-                 }
-                 break;
-             }
-+	    case kParamIndexMediaType:
```
Comment (basilchew) [inline bsp_diff/caas/vendor/intel/mediasdk_c2/0004-Fix-for-Vts-MultiRelease-decoder-TCs.patch:25]: I am unable to find the code or changes merged into horizontal.
Is this patch superceded by other changes?

**Review by basilchew — CHANGES_REQUESTED** [review]

File: bsp_diff/caas/device/intel/mixins/0031-Fix-CtsGraphicTest-testOpenGlEsDeqpLevel.patch [hunk 2]
```diff
@@ -1,28 +0,0 @@
...
-diff --git a/groups/graphics/auto/product.mk b/groups/graphics/auto/product.mk
-index 8bafbe0..e17d8dc 100755
---- a/groups/graphics/auto/product.mk
-+++ b/groups/graphics/auto/product.mk
-@@ -93,6 +93,9 @@ PRODUCT_COPY_FILES += \
- PRODUCT_COPY_FILES += \
-     frameworks/native/data/etc/android.hardware.opengles.aep.xml:vendor/etc/permissions/android.hardware.opengles.aep.xml
- 
-+PRODUCT_COPY_FILES += \
-+    frameworks/native/data/etc/android.software.opengles.deqp.level-2020-03-01.xml:vendor/etc/permissions/android.software.opengles.deqp.level.xml
```
[the remark this answers is not in the document]
Comment (vdanix) [reply bsp_diff/caas/device/intel/mixins/0031-Fix-CtsGraphicTest-testOpenGlEsDeqpLevel.patch:21]: > I traced the changes, and this one should be superceded by the patch:
> 
> * bsp_diff/caas/device/intel/mixins/0037-ADL-Fix-for-GTS-checkVulkanDeqpLevelIsHighEnough.patch
> * bsp_diff/caas/device/intel/project-celadon/0041-Mixin-up-patch-for-GTS-VulkanDeqpLevelIsHighEnough.patch
> * https://github.com/intel-innersource/os.android.bsp.vendor-intel-vertical-utils/pull/315/files
> 
> However, I am unable to find any Jira ticket that is merged into the horizontal, and I cannot find the final code in horizontal.
> 
> Please check.

Hi Basil,

Below are the Detail for Horizontal merge
Horizontal Jira : OAM-104806
Horizontal Merge PR :  https://github.com/projectceladon/device-androidia-mixins/pull/1560

File: bsp_diff/caas/device/intel/mixins/0037-ADL-Fix-for-GTS-checkVulkanDeqpLevelIsHighEnough.patch [hunk 3]
```diff
@@ -1,32 +0,0 @@
...
-diff --git a/groups/graphics/auto/product.mk b/groups/graphics/auto/product.mk
-index e17d8dc..0f893a2 100755
---- a/groups/graphics/auto/product.mk
-+++ b/groups/graphics/auto/product.mk
-@@ -94,7 +94,7 @@ PRODUCT_COPY_FILES += \
-     frameworks/native/data/etc/android.hardware.opengles.aep.xml:vendor/etc/permissions/android.hardware.opengles.aep.xml
- 
- PRODUCT_COPY_FILES += \
--    frameworks/native/data/etc/android.software.opengles.deqp.level-2020-03-01.xml:vendor/etc/permissions/android.software.opengles.deqp.level.xml
-+    frameworks/native/data/etc/android.software.opengles.deqp.level-2021-03-01.xml:vendor/etc/permissions/android.software.opengles.deqp.level.xml
```
[the remark this answers is not in the document]
Comment (vdanix) [reply bsp_diff/caas/device/intel/mixins/0037-ADL-Fix-for-GTS-checkVulkanDeqpLevelIsHighEnough.patch:17]: > I traced the changes, and this patch supercedes:
> 
> * bsp_diff/caas/device/intel/mixins/0031-Fix-CtsGraphicTest-testOpenGlEsDeqpLevel.patch
> * bsp_diff/caas/device/intel/project-celadon/0036-Fix-CtsGraphicTest-testOpenGlEsDeqpLevel.patch
> * https://github.com/intel-innersource/os.android.bsp.vendor-intel-vertical-utils/pull/288/files
> 
> However, I am unable to find any Jira ticket that is merged into the horizontal, and I cannot find the final code in horizontal.
> 
> Please check.

Hi Basil,

Below are the Detail for Horizontal merge
Horizontal Jira : OAM-104806
Horizontal Merge PR :  https://github.com/projectceladon/device-androidia-mixins/pull/1560

File: bsp_diff/caas/device/intel/mixins/0031-Fix-CtsGraphicTest-testOpenGlEsDeqpLevel.patch (hunk 2 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0031-Fix-CtsGraphicTest-testOpenGlEsDeqpLevel.patch:21]: Ok got it

File: bsp_diff/caas/device/intel/mixins/0037-ADL-Fix-for-GTS-checkVulkanDeqpLevelIsHighEnough.patch (hunk 3 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0037-ADL-Fix-for-GTS-checkVulkanDeqpLevelIsHighEnough.patch:17]: Ok got it

File: bsp_diff/caas/vendor/intel/mediasdk_c2/0004-Fix-for-Vts-MultiRelease-decoder-TCs.patch (hunk 1 above)
Comment (pchenthx1) [reply bsp_diff/caas/vendor/intel/mediasdk_c2/0004-Fix-for-Vts-MultiRelease-decoder-TCs.patch:25]: Yes, Basil. This is superseded by https://github.com/projectceladon/MediaSDK_C2/pull/75/
So, this patch is not relevant anymore

File: bsp_diff/caas/vendor/intel/mediasdk_c2/0002-Fix-for-VTS-MultiRelease-Timeout-encoder-TCs.patch [hunk 4]
```diff
@@ -1,36 +0,0 @@
...
-             continue;
-         }
-         // check whether plugin is in a correct state to apply this parameter
--        // the check is bypassed for bitrate parameter as it should be updatable
--        // despite its type 'info' (workaround of Google's bug)
-         bool modifiable = (param->coreIndex().coreIndex() == kParamIndexBitrate) ||
--            (param->kind() == C2Param::TUNING) || (param->kind() == C2Param::INFO) ||
--            (param->kind() == C2Param::SETTING && m_state == State::STOPPED);
-+		((param->kind() & C2Param::TUNING) != 0) ||
-+		m_state == State::STOPPED; /* all kinds, even INFO might be set by stagefright */
```
[the remark this answers is not in the document]
Comment (pchenthx1) [reply bsp_diff/caas/vendor/intel/mediasdk_c2/0002-Fix-for-VTS-MultiRelease-Timeout-encoder-TCs.patch:30]: Yes, Basil. This is superseded by https://github.com/projectceladon/MediaSDK_C2/pull/75/
So, this patch is not relevant anymore

File: bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch [hunk 5]
```diff
@@ -21,18 +24,14 @@ index f3d6c15..60da9e1 100755
...
- endif
-
+ 
 -ifeq ($(BASE_LTS2020_YOCTO_KERNEL),true)
 -BOARD_KERNEL_CMDLINE += i915.enable_guc=1
--endif
--
++ifeq ($(BASE_YOCTO_KERNEL),true)
++BOARD_KERNEL_CMDLINE += i915.enable_guc=2
+ endif
```
[the remark this answers is not in the document]
Comment (pchenthx1) [reply bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch]: Patch removed

File: bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch [hunk 6]
```diff
@@ -21,18 +24,14 @@ index f3d6c15..60da9e1 100755
...
  {{/acrn-guest}}
-
-@@ -13,10 +17,6 @@ ifeq ($(BASE_YOCTO_KERNEL),true)
- BOARD_KERNEL_CMDLINE += i915.enable_guc=2
- endif
-
+ 
 -ifeq ($(BASE_LTS2020_YOCTO_KERNEL),true)
 -BOARD_KERNEL_CMDLINE += i915.enable_guc=1
--endif
```
[the remark this answers is not in the document]
Comment (pchenthx1) [reply bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch:31]: Removed patch 0007-Enable-GUC-for-HWC-support.patch

File: bsp_diff/caas/device/intel/mixins/0028-Fix-to-enable-h-w-codec.patch [hunk 7]
```diff
@@ -12,14 +12,14 @@ Signed-off-by: gollarx <ratnakumarix.golla@intel.com>
...
-index c305683..21eed01 100644
+index 7d0fbb5..21eed01 100644
 --- a/groups/media/auto/auto_hal.in
 +++ b/groups/media/auto/auto_hal.in
 @@ -13,13 +13,8 @@ case "$(cat /proc/fb)" in
                  setprop vendor.intel.video.codec hardware
                  ;;
          *)
--                if [ "$(cat /sys/kernel/debug/dri/0/i915_sriov_info |grep virtualization |awk '{print $2}')" = "enabled" ];then
+-                if [ -d "/sys/class/drm/card0/engine" ];then
```
[the remark this answers is not in the document]
Comment (pchenthx1) [reply bsp_diff/caas/device/intel/mixins/0028-Fix-to-enable-h-w-codec.patch]: removed patch 0028-Fix-to-enable-h-w-codec.patch

File: bsp_diff/caas/device/intel/mixins/0030-Configure-rro-overlay-for-connectivity-to-fix-CTS-is.patch [hunk 8]
```diff
@@ -1,31 +0,0 @@
...
-     TetheringConfigOverlay \
--    TetheringConfigOverlayGsi
-+    TetheringConfigOverlayGsi \
-+    ServiceConnectivityResourcesConfigOverlay
- 
- PRODUCT_PACKAGES += \
-     android.hardware.wifi@1.0-service
--- 
-2.37.3
-
```
[the remark this answers is not in the document]
Comment (pchenthx1) [reply bsp_diff/caas/device/intel/mixins/0030-Configure-rro-overlay-for-connectivity-to-fix-CTS-is.patch:31]: removed patch vendor/intel/utils_vertical/bsp_diff/caas/device/intel/common/0002-Configure-rro-overlay-for-connectivity-to-fix-CTS-is.patch

File: bsp_diff/caas/vendor/intel/mediasdk_c2/0004-Fix-for-Vts-MultiRelease-decoder-TCs.patch (hunk 1 above)
Comment (basilchew) [reply bsp_diff/caas/vendor/intel/mediasdk_c2/0004-Fix-for-Vts-MultiRelease-decoder-TCs.patch:25]: Ok noted

File: bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch (hunk 6 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch:31]: Ok

File: bsp_diff/caas/device/intel/mixins/0030-Configure-rro-overlay-for-connectivity-to-fix-CTS-is.patch (hunk 8 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0030-Configure-rro-overlay-for-connectivity-to-fix-CTS-is.patch:31]: Ok

File: bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch (hunk 5 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0007-Enable-GUC-for-HWC-support.patch]: Ok

File: bsp_diff/caas/device/intel/mixins/0028-Fix-to-enable-h-w-codec.patch (hunk 7 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0028-Fix-to-enable-h-w-codec.patch]: Ok

File: bsp_diff/caas/vendor/intel/mediasdk_c2/0002-Fix-for-VTS-MultiRelease-Timeout-encoder-TCs.patch (hunk 4 above)
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/vendor/intel/mediasdk_c2/0002-Fix-for-VTS-MultiRelease-Timeout-encoder-TCs.patch:30]: Ok

File: bsp_diff/caas/device/intel/mixins/0024-Remove-clearcpuid.patch [hunk 9]
```diff
@@ -1,30 +1,27 @@
...
--ifeq ($(BASE_LTS2020_YOCTO_KERNEL), true)
--BOARD_KERNEL_CMDLINE += \
+ BOARD_KERNEL_CMDLINE += \
 -      clearcpuid=517 \
--      mce=no_lmce
--endif
--
+-      mce=no_lmce \
+       mce=ignore_ce
+
```
[the remark this answers is not in the document]
Comment (basilchew) [reply bsp_diff/caas/device/intel/mixins/0024-Remove-clearcpuid.patch]: This should be the last item.
If we remove 0024-Remove-clearcpuid.patch, does the kernel crashdump appear in the dmesg?

[the remark this answers is not in the document]
Comment (pchenthx1) [reply bsp_diff/caas/device/intel/mixins/0024-Remove-clearcpuid.patch]: Verified and no crash dump observed in dmesg during boot / shutdown

**Comment by pchenthx1** [comment]
@basilchew @bnkoppad please review

**Review by basilchew — APPROVED** [review]

**Pull request merged**